### DISCUSSION TITLE
ManualControlCapability for MQTT?

### DISCUSSION THREAD

**Discussion by APayerl** [Hypfer/Valetudo#1781] [Q&A / Support] [answered]
So I started looking at the API for REST and MQTT to learn how I could tell my vacuum to "take a step forward" so that I can empty its dust bin (my vacuum lives below a cupboard) on a button press from home assistant. 
After a while I realised that while it's possible using the “Valetudo Cloud”-app and using REST, it is not possible using MQTT since there is no `CapabilityMqttHandle` made for MQTT like it is for rest (`ManualControlCapabilityRouter`).
Is this a design choice or is it just a matter of too many plans and too little time?

**Comment by Hypfer** [comment, marked as answer]
To use the `ManualControlCapability`, you need a human with a direct line of sight to the robot. It is not possible to use that to reliably automate routes or any movement for that matter. The reason why it is part of the REST API is just because it needs to be so that it can be controlled via the Web UI. It is not meant for automation.

What you'd want here would be a robot that has a firmware that supports the `GoToLocationCapability`.
If your robot doesn't support that, some people also reported that they are using the `ZoneCleaningCapability` + a Home Assistant (or similar) automation to achieve somewhat similar results.

The basic idea there is:
1. Send clean Zone command with tiny zone where it should drive to
2. Wait for state change to returning
3. Send stop command

If that's something you'd want to implement in your home automation setup as well, you may want to consider doing it as a blueprint and opening a PR here: https://github.com/mundschenk-at/ha-valetudo-blueprints

**Reply by APayerl** [reply]
Thank you for your quick reply and the solution suggestion!
Probably I'll just use the Home Assistants rest_command integration instead to do the intended action.
